site/public/js/calendar.js

```diff
@@ -0,0 +1,320 @@
+/* exported prevMonth */
+/* exported nextMonth */
+/* exported loadCalendar */
+/* exported loadFullCalendar */
+/* global curr_day */
+/* global curr_month */
+/* global curr_year */
+/* global gradeables_by_date */
+
+// List of names of months in English
+const monthNames = ['December', 'January', 'February', 'March', 'April', 'May', 'June', 'July', 'August', 'September', 'October', 'November', 'December'];
+const monthNamesShort = ['Dec', 'Jan', 'Feb', 'Mar', 'Apr', 'May', 'Jun', 'Jul', 'Aug', 'Sept', 'Oct', 'Nov', 'Dec'];
+
+/**
+ * Gets the previous month of a given month
+ * @param month : int the current month (1 as January and 12 as December)
+ * @param year : int the current year
+ * @returns {number[]} : array<int> {previous_month, year_of_previous_month}
+ */
+function prevMonth(month, year) {
+    month = month - 1;
+    if (month <= 0) {
+        month = 12 + month;
+        year = year - 1;
+    }
+    return [month, year];
+}
+
+/**
+ * Gets the next month of a given month
+ *
+ * @param month : int the current month (1 as January and 12 as December)
+ * @param year : int the current year
+ * @returns {number[]} : array<int> {next_month, year_of_next_month}
+ */
+function nextMonth(month, year) {
+    month = month + 1;
+    if (month > 12) {
+        month = month - 12;
+        year = year + 1;
+    }
+    return [month, year];
+}
+
+
+/**
+ * This function creates a Date object based on a string.
+ *
+ * @param datestr : string a string representing a date in the format of YYYY-mm-dd
+ * @returns {Date} a Date object containing the specified date
+ */
+function parseDate(datestr){
+    const temp = datestr.split('-');
+    return new Date(temp[0], temp[1]-1, temp[2]);
+}
+
+/**
+ * This function creates a string in the format of YYYY-mm-dd.
+ *
+ * @param year : int the year
+ * @param month : int the month
+ * @param day : int the date
+ * @returns {string}
+ */
+function dateToStr(year, month, day) {
+    return `${year.toString()}-${month.toString().padStart(2, '0')}-${day.toString().padStart(2, '0')}`;
+}
+
+/**
+ * Creates a HTML table cell that contains a date.
+ *
+ * @param year : int the year of the date
+ * @param month : int the month of the date (1 as January and 12 as December)
+ * @param day : int the date of the date (1 - 31)
+ * @param curr_view_month : int the current month that the calendar is viewing
+ * @param view_semester : if the calendar is viewing the entire semester. If so, the day cell would show both the month and date
+ * @returns {string} the HTML string containing the cell
+ */
+function getDayCell(year, month, day, curr_view_month, view_semester=false) {
```

Comment:
Nit: `getDayCell` seems to imply we’re fetching data from somewhere rather than constructing a blob of HTML to output somewhere else, `generateDayCell` seems like a more appropriate name for this